lib/pf/SwitchFactory.pm

```diff
@@ -134,23 +149,23 @@ sub instantiate {
 
     # find the module to instantiate
     my $switchOverlay = $switch_overlay_cache->get($requestedSwitch) || {};
-    my $type;
+    my $module;
+    my $type = $switch_data->{'type'};
     if ($requestedSwitch ne 'default') {
-        $type = "pf::Switch::" . $switch_data->{'type'};
+        $module = getModule($type);
     } else {
-        $type = "pf::Switch";
+        $module = "pf::Switch";
     }
-    $type = untaint_chain($type);
-    # load the module to instantiate
-    if ( !(eval "$type->require()" ) ) {
+    unless ($module) {
         $logger->error("Can not load perl module for switch $requestedSwitch, type: $type. "
-            . "Either the type is unknown or the perl module has compilation errors. "
-            . "Read the following message for details: $@");
-        return 0;
+                  . "The type is unknown or the perl module has compilation errors. ");
+        return undef;
```

Comment:
Really not sure about this one. When we instanciate we then test the result with if(!$switch) which will warn for undefined value. That's what I think but I may be wrong.